api/v1alpha1/envoygateway_types.go

```diff
@@ -88,6 +90,20 @@ type EnvoyGatewaySpec struct {
 	//
 	// +optional
 	ExtensionAPIs *ExtensionAPISettings `json:"extensionApis,omitempty"`
+
+	// LeaderElection defines the settings related to leader election
+	// +optional
+	LeaderElection *LeaderElection `json:"leaderElection,omitempty"`
```

Comment:
just realized this lives in the top level today, this needs to be moved into the kubernetes specific provider field